Modules/Core/Common/include/itkWeakPointer.h

```diff
@@ -47,23 +47,21 @@ class WeakPointer
   /** Extract information from template parameter. */
   using ObjectType = TObjectType;
 
-  /** Constructor.  */
-  WeakPointer() { m_Pointer = nullptr; }
+  /** Explicitly-defaulted default-constructor.
+   * \note The other five "special member functions" (copy-constructor,
+   * copy-assignment operator, move-constructor, move-assignment operator,
+   * and destructor) are defaulted implicitly, following the C++ "Rule of Zero".
```

Comment:
Thanks for your approval, @hjmjohnson Do you think it's still useful to explain what is meant by "special member functions", like I'm doing here? I meant, is it still useful to mention those five member functions explicitly here?

    copy-constructor, copy-assignment operator, move-constructor, move-assignment operator, and destructor

I feel that the term "special member functions" is a bit "special" (possibly not familiar to everyone), that's why I did. But what do you think?